pkg/asset/installconfig/azure/session.go

```diff
@@ -43,8 +43,8 @@ type Credentials struct {
 	ClientID                  string `json:"clientId,omitempty"`
 	ClientSecret              string `json:"clientSecret,omitempty"`
 	TenantID                  string `json:"tenantId,omitempty"`
-	ClientCertificatePath     string `json:"certificatePath,omitempty"`
-	ClientCertificatePassword string `json:"certificatePassword,omitempty"`
+	ClientCertificatePath     string `json:"clientCertificate,omitempty"`
```

Comment:
Wasn't `certificatePath and certificatePassword` how azure expects certificates to be populated in the `osServicePrincipal.json` file? 

Even if the installer does not need it with that key name, we should have the same name as azure expects so it makes it easier for users to use the json file for both azure auth(like az client) and the installer without change.

Reply:
autorest/auth was the lib that was loading that file, but I'm removing it with this PR since it's going out of support by the end of March. I've actually just changed it to the same values the lib was expecting https://github.com/Azure/go-autorest/blob/9038e4a609b1899f0eb382d03c3e823b70537125/autorest/azure/auth/auth.go#L348-L349

Reply:
We do need to keep backwards compatibility for existing users though; and this change would break anyone who is currently authenticating with certs. 

We could potentially support both fields, but it may not be worth it considering the limited scope of this feature.

Reply:
Just users using the old json file need to change the key name for these fields. If they choose to update the installer version they have to also change this key which they won't know of.

Reply:
If they do, then this never worked in the first place. `autorest/auth` was loading the json file, and it expects the `clientCertificate`  and `clientCertificatePassword` names.

Reply:
See https://github.com/Azure/go-autorest/blob/main/autorest/azure/auth/auth.go#L248-L249 which calls https://github.com/Azure/go-autorest/blob/main/autorest/azure/auth/auth.go#L321-L349

Reply:
These fields in Credentials would have been used if we ever asked for Certificate details during the survey. That's the only time when we use that json annotation to save the Credentials object to a file.

Reply:
You're right, looks like before this PR, it could be anything since we don't use it to write to file since we don't collect certs in survey. Shouldn't have a problem with backward compatibility.

Reply:
Given how confusing this change is, maybe I should put it in its own commit with a reason why it's needed?

Reply:
I was mistaken/confused in my original comment and I agree this does not break backward compatibility. I missed that these JSON tags are only used for serialization--not deserialization. By that same token, it seems like we could just remove the json tags for the cert fields, right? 

But this is definitely fine with me as it is.

Reply:
We cannot remove them because with this PR they **are** used for deserialization now. That was the point of changing the annotation to be compatible with what autorest expected.

Reply:
:man_facepalming: